### DISCUSSION TITLE
Submit default form values on reset

### DISCUSSION THREAD

**Discussion by Moshyfawn** [react-hook-form/react-hook-form#6058] [Q&A] [answered]
Can you advise on how to submit values on from reset? Think clear filters button

**Comment by bluebill1049** [comment]
```import React from "react";
import { useForm, useFieldArray, Controller } from "./src";

function App() {
  const {
    register,
    handleSubmit,
    reset,
    formState: { isSubmitSuccessful }
  } = useForm({ defaultValues: { something: "anything" } });

  const onSubmit = (data) => {
    console.log(data);
  };

  React.useEffect(() => {
    if (formState.isSubmitSuccessful) {
      reset({ something: '' });
    }
  }, [formState, submittedData, reset]);

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <input {...register("something")} />
      <input type="submit" />
    </form>
  );
}
```

<img width="1008" alt="Screen Shot 2021-07-26 at 11 24 35 am" src="https://user-images.githubusercontent.com/10513364/126921086-485deabc-ebf1-4ffa-a86a-0d869ce89a74.png">

**Comment by Moshyfawn** [comment, marked as answer]
Hey, @bluebill1049!
Thx for your time!
First of all, sry for a non-detailed question: I was writing from my phone on the way with an idea to add some additional info later. _(obvsly forgot)_
I've seen the example from the [RHF documentation](https://react-hook-form.com/api/useform/reset/#main) you're referring to. While it suites the basic submit -> reset form values pattern, in my case I want to either submit or reset a form value independently and proceed to use the form value after.
What I was looking for is to have a callback on form reset, but couldn't figure it out for some reason. What I ended up with is to utilize [an onReset form method](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onreset).
See the example codesanbox here: https://codesandbox.io/s/busy-worker-lcm4w?file=/src/App.tsx
